x/auctions/keeper/hooks_test.go

```diff
@@ -0,0 +1,234 @@
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Params:    params,
+				}
+				_, err := network.App.AuctionsKeeper.UpdateParams(network.GetContext(), &updateParamsMsg)
+				assert.NoError(t, err, "failed to update auctions params")
+			},
+			expUpdate: false,
+			postCheck: func() {},
+		},
+		{
+			name: "no op if not valid bidder address",
+			malleate: func() {
+				setHighestBid(t, network, "", bidAmount)
+			},
+			expUpdate: false,
+			postCheck: func() {},
+		},
+		{
+			name: "no op if bid amount is not positive",
+			malleate: func() {
+				setHighestBid(t, network, validSenderAddr.String(), sdk.NewInt64Coin(utils.BaseDenom, 0))
+			},
+			expUpdate: false,
+			postCheck: func() {},
+		},
+		{
+			name: "no op if auctions module does not holds the bid amount",
+			malleate: func() {
+				err := network.App.BankKeeper.BurnCoins(network.GetContext(), types.ModuleName, sdk.NewCoins(bidAmount))
+				assert.NoError(t, err, "failed to burn coins in malleate")
+			},
+			expUpdate: false,
+			postCheck: func() {},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(_ *testing.T) {
+			keyring := testkeyring.New(1)
+			network = testnetwork.NewUnitTestNetwork(
+				testnetwork.WithPreFundedAccounts(keyring.GetAllAccAddrs()...),
+			)
+
+			epochIdentifier = epochstypes.WeekEpochID
+
+			// Following code define the chain state required for an auction to be
+			// performed properly. malleate() function modify the state to test all
+			// possible cases.
+			// Check that we are at the first round.
+			round := network.App.AuctionsKeeper.GetRound(network.GetContext())
+			assert.Equal(t, uint64(0), round, "round different than initial one")
+
+			// Inflation should be enabled.
+			params := network.App.AuctionsKeeper.GetParams(network.GetContext())
+			params.EnableAuction = true
+			updateParamsMsg := types.MsgUpdateParams{
+				Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+				Params:    params,
+			}
+			_, err := network.App.AuctionsKeeper.UpdateParams(network.GetContext(), &updateParamsMsg)
+			assert.NoError(t, err, "failed to update auctions params")
+
+			// There should be an highest bid.
+			setHighestBid(t, network, validSenderAddr.String(), bidAmount)
+
+			// We have to mint the bidded coins and also additional ones that will be distributed.
+			mintCoinsToModuleAccount(t, network, types.ModuleName, sdk.NewCoins(append(additionalCoins, bidAmount)...))
+
+			// We have to mint coins for the auction collector to simulate accrued fees that
+			// are then sent from the auction collector to the auctions module.k
+			mintCoinsToModuleAccount(t, network, types.AuctionCollectorName, sdk.NewCoins(feeCoin))
+
+			tc.malleate()
+			ctxSnapshot := network.GetContext()
+			network.App.AuctionsKeeper.AfterEpochEnd(network.GetContext(), epochIdentifier, unusedEpochNumber)
+
+			tc.postCheck()
+			if tc.expUpdate {
+				tc.postCheck()
+			} else {
```

Comment:
post check is run twice here? anyways as mentioned on other comments, I don't think we should have a "hidden" postcheck but rather have the logic here -- especially since the logic for all of the testcases above is the same logic as is tested below, just with different numbers